File: HelpSoftware/Help/EF/MailActivity.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Help
{
  public class MailActivity
  {
    [Key]
    public int ID_MailActivity { get; set; }

    public DateTime SendTime { get; set; }

    public string Sender { get; set; }

    public User UserSender { get; set; }

    public string Subject { get; set; }

    public string Recipient { get; set; }

    public string MailType { get; set; }
  }
}

File: HelpSoftware/Help/Help.Web/Web/src/app/EF/MailActivity.ts
import { User } from "./EF";

export class MailActivity {
  ID_MailActivity: number;
  SendTime: Date;
  Sender: string;
  UserSender: User;
  Subject: string;
  Recipient: string;
  MailType: string;

  deserialize(input): MailActivity {
    this.ID_MailActivity = input.ID_MailActivity;
    this.SendTime = input.SendTime;
    this.Sender = input.Sender;
    this.UserSender = new User().deserialize(input.UserSender);
    this.Subject = input.Subject;
    this.Recipient = input.Recipient;
    this.MailType = input.MailType;
    return this;
  }
}
